### DISCUSSION TITLE
[External project "assimp"] Loading of some models broken due to early return introduced at commit 30481ec

### DISCUSSION THREAD

**Discussion by tellypresence** [syoyo/tinyusdz#176] [Q&A] [answered]
Fixed by #177 

(This regards `tinyusdz` integration in external project `assimp`)

_Bisected and found the 2 adjacent commits where it was working (`616302d`), then broke ([`30481ec`](https://github.com/lighttransport/tinyusdz/commit/30481ec2fad7c3643c78e1760e84900688d50531))_

2 models downloaded from Sketchfab were working up through commit `616302d` back in April, then broke at the next commit [`30481ec`](https://github.com/lighttransport/tinyusdz/commit/30481ec2fad7c3643c78e1760e84900688d50531).  

The cause was an early return introduced in file `src/tydra/render-data.cc` at commit [`30481ec`](https://github.com/lighttransport/tinyusdz/commit/30481ec2fad7c3643c78e1760e84900688d50531)

Replacing the early return
```
  if (skelPath.is_valid()) {
    if (!ConvertSkeletonImpl(env, abs_path.full_path_name(), mesh, &skel)) {
      return false; // <<< this will never continue additional crucial model processing
    }
    // Do skeleton stuff
  }
  // Do additional crucial model processing
```
with opposite logic so skeleton processing done if check succeeds
```
  if (skelPath.is_valid()) {
    if (ConvertSkeletonImpl(env, abs_path.full_path_name(), mesh, &skel)) {
        // Do skeleton stuff
    }
  }
  // Do additional crucial model processing
```
fixes the problem; the breakage was due to other crucial steps needing to be done in the same function after the skeleton processing -- this change makes skeleton steps optional instead of aborting the model processing if skeleton check fails.

<table>
<tr><th>Commit</th><th>Model</th><th>Screenshot</th><th>Notes</th></tr>
<tr><td rowspan=2><a href="https://github.com/lighttransport/tinyusdz/commit/30481ec2fad7c3643c78e1760e84900688d50531"><tt>30481ec</tt></a><br>20 Apr 2024<br>Broken</td><td><a href="https://sketchfab.com/3d-models/medical-assistance-bot-cb494908e11543699d682bb2c71f0817"><tt>Medical Assistance Bot</tt></a></td><td>N/A</td><td></td></tr>
<tr><td><a href="https://sketchfab.com/3d-models/exgundam-ex-120c475ecf10426e802ce9722b31f3f3"><tt>ガンダムEX/Gundam EX</tt></a>
</td><td>N/A</td><td></td></tr>
<tr><td rowspan=2><a href="https://github.com/lighttransport/tinyusdz/commit/616302d5bf63f905fe9186cceb036dcb3bbbb6a0"><tt>616302d</tt></a><br>20 Apr 2024<br>Working</td><td><a href="https://sketchfab.com/3d-models/medical-assistance-bot-cb494908e11543699d682bb2c71f0817"><tt>Medical Assistance Bot</tt></a></td><td><img src="https://github.com/user-attachments/assets/875e3c26-fe0d-44a8-ab78-5ea92f6c34bb" alt="Medical Assistance Bot" width=180 /></td><td></td></tr>
<tr><td><a href="https://sketchfab.com/3d-models/exgundam-ex-120c475ecf10426e802ce9722b31f3f3"><tt>ガンダムEX/Gundam EX</tt></a></td><td><img src="https://github.com/user-attachments/assets/6acf29b5-018d-4d42-a6c4-dd58e610ad5c" alt="ガンダムEX/Gundam EX" width=180 /></td><td></td></tr>
</table>

**Comment by tellypresence** [comment, marked as answer]
Source of problem has been determined via bisect and iteration

**Comment by syoyo** [comment]
You should not post a discussion about external project.
